src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -416,15 +416,25 @@ public int GetHashCode(Symbol member)
                     // CONSIDER: could use interface type, but that might be quite expensive
                 }
 
-                if (_considerReturnType && member.GetMemberArity() == 0 && !_considerCustomModifiers) // If it is generic, then type argument might be in return type.
+                if (_considerReturnType)
                 {
-                    hash = Hash.Combine(member.GetTypeOrReturnType(), hash);
+                    RefKind refKind;
+                    TypeSymbol returnType;
+                    ImmutableArray<CustomModifier> returnTypeCustomModifiers;
+                    member.GetTypeOrReturnType(out refKind, out returnType, out returnTypeCustomModifiers);
+
+                    hash = Hash.Combine((int)refKind, hash);
+
+                    if (member.GetMemberArity() == 0 && !_considerCustomModifiers) // If it is generic, then type argument might be in return type.
+                    {
+                        hash = Hash.Combine(returnType, hash);
```

Comment:
It looks like returnType can contain custom modifiers, it is not clear why it is the right thing to get here when _considerCustomModifiers is false.